tests/robustness/robustness_test/main_test.go

```diff
@@ -48,7 +48,14 @@ func TestMain(m *testing.M) {
 	eng = th.engine
 
 	// Restore a random snapshot into the data directory
-	if _, err := eng.ExecAction(ctx, engine.RestoreIntoDataDirectoryActionKey, nil); err != nil && !errors.Is(err, robustness.ErrNoOp) {
+	if opts, err := eng.ExecAction(ctx, engine.RestoreIntoDataDirectoryActionKey, nil); err != nil && !errors.Is(err, robustness.ErrNoOp) {
+		if eng.ErrIsNotEnoughSpace(err) {
+			// Delete the snapshot
+			snapID := opts["snapshot-ID"]
+			log.Println("deleting snap ID: ", snapID)
+			_ = eng.Checker.DeleteSnapshot(ctx, snapID, opts)
```

Comment:
Is this deleting the snapshot in the repository?

Reply:
Good point. The `eng.ExecAction` and `eng.Checker.xx` may be acting upon two different repositories (object store and local). Let me confirm this.